app/models/service_template_ansible_tower.rb

```diff
@@ -24,7 +24,7 @@ def self.create_catalog_item(options, _auth_user = nil)
 
   def remove_invalid_resource
     # remove the resource from both memory and table
-    service_resources.to_a.delete_if { |r| r.destroy unless r.resource(true) }
```

Comment:
@bzwei Can you review this change?  I needed to remove the `.resource(true)` to force reload of the association because it was removed in rails 5.1.  The alternative of `.reload.resource` causes issues because this `before_save` is called before creates and updates and blows up on creates because it's not yet persisted.  I changed this to a before_update, to simplify the code based on the comment you added.

Reply:
I understand the `s/\](true\)/reload`

curious:

Do we want to change from `delete_if` to `each`? think the `to_a` negates any real side effects.